File: 40_dynamic_programming/climbing_stairs.py
def climbing_stairs(n: int)->int:
  dp = [0] * n
  dp[0] = 1
  dp[1] = 2

  for i in range(2, n):
    dp[i] = dp[i-1] + dp[i-2]
  
  return dp[-1]

if __name__ == "__main__":
  n = 4
  print(climbing_stairs(n))

File: 40_dynamic_programming/climbing_stairs.cpp
#include <iostream>
#include <vector>

int climbing_stairs(int n) {
  std::vector<int> dp(n, 0);
  dp[0] = 1;
  dp[1] = 2;

  for (int i=2; i < n; i++) {
    dp[i] = dp[i-1] + dp[i-2];
  }

  return dp[n-1];
}

int main() {
  int n = 4;
  std::cout << climbing_stairs(n) << std::endl;
  return 0;
}